packages/playwright-core/src/server/dispatchers/pageDispatcher.ts

```diff
@@ -306,19 +308,23 @@ export class PageDispatcher extends Dispatcher<Page, channels.PageChannel, Brows
   async startJSCoverage(params: channels.PageStartJSCoverageParams, metadata: CallMetadata): Promise<void> {
     const coverage = this._page.coverage as CRCoverage;
     await coverage.startJSCoverage(params);
+    this._jsCoverageActive = true;
```

Comment:
Set it to true in the beginning, so that onDispose calls stop even if startJSCoverage was not awaited?